flo-freezer/src/main/java/com/spotify/flo/freezer/PersistingContext.java

```diff
@@ -180,4 +176,32 @@ public static String cleanForFilename(TaskId taskId) {
   private Path taskFile(TaskId taskId) {
     return basePath.resolve(cleanForFilename(taskId));
   }
+
+  private static boolean hasChillScala() {
+    try {
+      ScalaKryoInstantiator.class.getName();
+      return true;
+    } catch (NoClassDefFoundError e) {
+      LOG.debug("Could not find com.twitter.chill.ScalaKryoInstantiator: {}", e.toString());
+      return false;
+    }
+  }
+
+  private static class ScalaKryoSetup {
```

Comment:
What's the benefit having a static class here instead of just a static method? And why is the branch `using chill-java` inlined instead of being factored out?

Reply:
It needs to be a static class because we don't want the method to be class-loaded before we know that we have chill-scala available.

Reply:
https://github.com/spotify/flo/pull/174/files#diff-d7f1d9c331c8fb2c839cdd4ff6cb0a16R135